integration/test_ml.py

```diff
@@ -0,0 +1,361 @@
+# Copyright 2020 Google Inc.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+"""Integration tests for firebase_admin.ml module."""
+import re
+import os
+import shutil
+import random
+import tempfile
+import pytest
+
+
+from firebase_admin import ml
+from firebase_admin import exceptions
+from tests import testutils
+
+
+# pylint: disable=import-error,no-name-in-module
+try:
+    import tensorflow as tf
+    _TF_ENABLED = True
+except ImportError:
+    _TF_ENABLED = False
+
+
+NAME_ONLY_ARGS = {
+    'display_name': 'TestModel123_{0}'.format(random.randint(1111, 9999))
+}
+NAME_ONLY_ARGS_UPDATED = {
+    'display_name': 'TestModel123_updated_{0}'.format(random.randint(1111, 9999))
+}
+NAME_AND_TAGS_ARGS = {
+    'display_name': 'TestModel123_tags_{0}'.format(random.randint(1111, 9999)),
+    'tags': ['test_tag123']
+}
+FULL_MODEL_ARGS = {
+    'display_name': 'TestModel123_full_{0}'.format(random.randint(1111, 9999)),
+    'tags': ['test_tag567'],
+    'file_name': 'model1.tflite'
+}
+INVALID_FULL_MODEL_ARGS = {
+    'display_name': 'TestModel123_invalid_full_{0}'.format(random.randint(1111, 9999)),
+    'tags': ['test_tag890'],
+    'file_name': 'invalid_model.tflite'
+}
+
+@pytest.fixture
+def firebase_model(request):
+    args = request.param
+    tflite_format = None
+    file_name = args.get('file_name')
+    if file_name:
+        file_path = testutils.resource_filename(file_name)
+        source = ml.TFLiteGCSModelSource.from_tflite_model_file(file_path)
```

Comment:
I assume each invocation of this makes an GCS upload? Do we need to clean them up at some point?

Reply:
No. They overwrite and they are not randomized. So there should only ever be 3 files.